minio/minio
# Changes to Access Key behavior when LDAP is configured then disabled

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
Add check in CheckKey that returns `!ok` for LDAP access key when LDAP is disabled

If LDAP config is deleted, `purgeLDAPAccessKeys` runs on server startup to delete any access keys for LDAP accounts

## Motivation and Context
Currently, access keys (both STS and service accounts) for LDAP accounts continue to function even if LDAP configuration has been disabled or removed. We would like to change this behavior different ways depending on scenario:

- If LDAP has been disabled but the configuration not removed, we would like access keys to not work. However, if the configuration is re-enabled, these access keys should begin to work again
- If LDAP has been deleted, all LDAP-associated access keys should be deleted

## How to test this PR?
Enable LDAP and create STS or service accounts for LDAP users. After disabling LDAP, these accounts should not work. Re-enable LDAP, and these accounts should function as normal. Remove LDAP configuration. After re-adding the config, access keys should no longer exist. 

Note that LDAP users in groups get errors if they try to log in when LDAP is enabled currently. To properly show change of PR, test with an LDAP user not in a group, like `ivanov` in `minio-iam-testing`

## Types of changes

## Checklist:
